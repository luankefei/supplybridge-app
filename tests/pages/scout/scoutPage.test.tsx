import React from "react";
import { render, screen } from "@testing-library/react";
import Scout from "pages/scout";
import { renderWithTheme } from "../../test-utils";

// This has to be mocked before running the test
jest.mock("next/router", () => require("next-router-mock"));
// mock the map
/**
 * Unfortunately, Jest cannot natively handle these kinds of tests as it does not have * full DOM or SVG support.
 */
jest.mock("react-simple-maps", () => ({
  ComposableMap: (props: any) => <div>ComposableMap</div>,
  ZoomableGroup: (props: any) => <div>ZoomableGroup</div>,
  Geographies: (props: any) => <div>Geographies</div>,
  Geography: (props: any) => <div>Geography</div>,
  Marker: (props: any) => <div>Marker</div>,
}));

describe("Scout", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderWithTheme(<Scout />);
    expect(asFragment()).toMatchSnapshot();
  });
});
