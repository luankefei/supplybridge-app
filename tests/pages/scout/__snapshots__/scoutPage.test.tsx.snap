// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scout matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-hboir5"
  >
    <div
      class="MuiDrawer-root MuiDrawer-docked css-1hpp7ia-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-19n6fl3-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          class="css-2j8t8r-MuiStack-root"
        >
          <div
            class="css-1djdum4-MuiStack-root"
          >
            <picture>
              <img
                alt="logo"
                src="/favicon-32x32.png"
              />
            </picture>
            <div
              class="sc-dLMFU kRAazT"
            />
            <div
              class="MuiBox-root css-0"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uym7v6-MuiSvgIcon-root"
                  data-testid="ExpandIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 20h16v2H4zM4 2h16v2H4zm9 7h3l-4-4-4 4h3v6H8l4 4 4-4h-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <span
                class="sc-cKXybt jcjTjb"
              >
                SOL
              </span>
              <a
                href="/scout"
              >
                <div
                  class="sc-pKqro dBDOjl"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/scouting.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href="/source"
              >
                <div
                  class="sc-pKqro dBDOjl"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/sourcing.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href=""
              >
                <div
                  class="sc-pKqro hTXlRZ"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/evaluate.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href=""
              >
                <div
                  class="sc-pKqro hTXlRZ"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/sustainability.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href=""
              >
                <div
                  class="sc-pKqro hTXlRZ"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/finance.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href=""
              >
                <div
                  class="sc-pKqro hTXlRZ"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/transport.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href="/total-solution"
              >
                <div
                  class="sc-pKqro dBDOjl"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/total-solution.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
            </div>
            <div
              class="sc-dLMFU kSioOl"
            />
            <div
              class="MuiBox-root css-0"
            >
              <span
                class="sc-cKXybt jcjTjb"
              >
                MD
              </span>
              <a
                href="/raw-material"
              >
                <div
                  class="sc-pKqro dBDOjl"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/raw-material.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href="/supplier-news"
              >
                <div
                  class="sc-pKqro dBDOjl"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/supplier-news.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href="/supply-chain-transparency"
              >
                <div
                  class="sc-pKqro dBDOjl"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/supply-chain-transparency.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
            </div>
          </div>
          <div
            class="MuiBox-root css-1haa2nn"
          >
            <div
              class="css-rup1zz-MuiStack-root"
            >
              <a
                href="/notification"
              >
                <div
                  class="sc-pKqro dBDOjl"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/bell.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
              <a
                href="/account"
              >
                <div
                  class="sc-pKqro dBDOjl"
                >
                  <span
                    class="MuiBadge-root MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <img
                      class="sc-lgjHQU kYCZWi"
                      src="/menu/profile.svg"
                    />
                    <span
                      class="MuiBadge-badge MuiBadge-dot MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError MuiBadge-badge MuiBadge-invisible css-1iqv1uc-MuiBadge-badge"
                    />
                  </span>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <main
      class="MuiBox-root css-deuk67"
    >
      <div
        class="MuiBox-root css-1ukv10g"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="sc-bWJUgm hvGXME"
          >
            <div
              class="sc-kpKSZj ctgsqz"
            >
              <div
                class="sc-lfYzqA NdSLT"
              />
              <span
                class="sc-ijDOKB kZPSIl"
              >
                Scout by Index
              </span>
              <span
                class="sc-eBwKMn hxSexy"
              >
                Scout by QuickBridge
              </span>
              <span
                class="sc-bHnlcS bYuvau"
              >
                Bidders List
              </span>
            </div>
          </div>
          <div
            class="css-nen11g-MuiStack-root"
          >
            <div
              class="css-omazcq-MuiStack-root"
            >
              <div
                class="MuiBox-root css-42igfv"
              >
                <div
                  class="sc-dLMFU indsiC"
                />
                <span
                  class="css-1fmgu67"
                >
                  Global Scouting, for Automotive professionals.
                </span>
                <div
                  class="css-bimbu8"
                  style="margin: auto;"
                >
                  <img
                    class="sc-cPiKLX jfIsET"
                    height="30"
                    src="/icons/smart-bridge-ai.svg"
                    title=""
                    width="24"
                  />
                  <div
                    class="sc-eDPEul gJxffd"
                  />
                  <div
                    class="css-1lpx3xe"
                    color="#89A8B3"
                    font-size="13px"
                    font-weight="400"
                  >
                    Powered by SmartBridge Artificial Intelligence
                  </div>
                </div>
              </div>
              <div
                class="css-ed76l1-MuiStack-root"
              >
                <div
                  class="MuiBox-root css-ckyhk3"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ku2602-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <img
                      class="sc-cPiKLX gSEKAE"
                      height="12"
                      src="/icons/reset.svg"
                      title=""
                      width="12"
                    />
                    Reset
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="sc-fHjqPf iZczJw"
                  width="100"
                >
                  <div
                    class="sc-jsJBEP hJjHWK"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary sc-eeDRCY hklmpY css-1xutqql-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                    >
                      <div
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-labelledby="search_type"
                        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-sz0awi-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                        id="search_type"
                        role="button"
                        tabindex="0"
                      >
                        Keywords
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                        tabindex="-1"
                        value="Keywords"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-ihdtdm"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                    <img
                      class="sc-cPiKLX kpRJxm"
                      height="20"
                      src="/icons/search2.svg"
                      title=""
                      width="20"
                    />
                    <div
                      class="sc-eDPEul gJxffd"
                    />
                    <form
                      autocomplete="off"
                      style="width: 100%;"
                    >
                      <div
                        class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon sc-hzhJZQ dRKtxx css-1qqsdnr-MuiAutocomplete-root"
                      >
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1bpq8ua-MuiFormControl-root-MuiTextField-root"
                        >
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-17ahhef-MuiInputBase-root-MuiOutlinedInput-root"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-expanded="false"
                              aria-invalid="false"
                              autocapitalize="none"
                              autocomplete="off"
                              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-fr7zp0-MuiInputBase-input-MuiOutlinedInput-input"
                              id=":r0:"
                              placeholder="..."
                              role="combobox"
                              spellcheck="false"
                              type="text"
                              value=""
                            />
                            <div
                              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                            >
                              <button
                                aria-label="Clear"
                                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1glvl0p-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
                                tabindex="-1"
                                title="Clear"
                                type="button"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                                  data-testid="CloseIcon"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                  />
                                </svg>
                                <span
                                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                                />
                              </button>
                            </div>
                            <fieldset
                              aria-hidden="true"
                              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                            >
                              <legend
                                class="css-ihdtdm"
                              >
                                <span
                                  class="notranslate"
                                >
                                  ​
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                        </div>
                      </div>
                      <button
                        style="display: none;"
                        type="submit"
                      />
                    </form>
                  </div>
                  <div
                    class="sc-hmdomO jqEVRc"
                  >
                    <button
                      class="sc-bXCLTC fpGURK"
                    >
                      Search
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="sc-dLMFU kSPyXv"
              />
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiBox-root css-1ae5hgr"
                >
                  <div>
                    ComposableMap
                  </div>
                  <div
                    class="css-1w69ric-MuiStack-root"
                  >
                    <div
                      class="css-cu76v7"
                    >
                      <div
                        class="css-t2c9vy"
                      >
                        <div
                          class="MuiBox-root css-vsz5gp"
                        >
                          Total Results
                        </div>
                      </div>
                      <div
                        class="css-1nuqwwm"
                      >
                        <div
                          class="css-1tt9794"
                        >
                          <div
                            class="css-ak39sn"
                          >
                            Americas
                          </div>
                          <div
                            class="css-v5odcz"
                          >
                            0
                          </div>
                        </div>
                        <div
                          class="css-1tt9794"
                        >
                          <div
                            class="css-ak39sn"
                          >
                            EMEA
                          </div>
                          <div
                            class="css-v5odcz"
                          >
                            0
                          </div>
                        </div>
                        <div
                          class="css-1tt9794"
                        >
                          <div
                            class="css-ak39sn"
                          >
                            APAC
                          </div>
                          <div
                            class="css-v5odcz"
                          >
                            0
                          </div>
                        </div>
                      </div>
                      <div
                        class="css-g05or3"
                      >
                        <div
                          class="MuiBox-root css-1h7xuws"
                        >
                          <span>
                            Total
                          </span>
                          <span>
                            0
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
